WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/usecases/GetBackupDownloadStatusUseCase.kt

```diff
@@ -24,29 +24,34 @@ import javax.inject.Named
 
 const val DELAY_MILLIS = 1000L
 const val MAX_RETRY = 3
+const val DELAY_FACTOR = 2
 
 class GetBackupDownloadStatusUseCase @Inject constructor(
     private val networkUtilsWrapper: NetworkUtilsWrapper,
     private val activityLogStore: ActivityLogStore,
     @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
 ) {
+    private val tag = javaClass.simpleName
     suspend fun getBackupDownloadStatus(
         site: SiteModel,
         downloadId: Long? = null
     ) = flow {
-        val tag = javaClass.simpleName
         var retryAttempts = 0
+        var backoffDelay = DELAY_MILLIS
         while (true) {
             if (!isNetworkAvailable()) return@flow
 
             val result = activityLogStore.fetchBackupDownloadState(FetchBackupDownloadStatePayload(site))
             if (result.isError) {
-                if (retryAttempts++ >= MAX_RETRY) {
-                    AppLog.d(T.JETPACK_BACKUP, "$tag Exceeded 3 retries while fetching status")
-                    emit(RemoteRequestFailure)
-                    return@flow
+                when (exceedsRetryAttempts(retryAttempts++)) {
+                    true -> return@flow
+                    false -> {
+                        delay(backoffDelay)
+                        backoffDelay = (backoffDelay * DELAY_FACTOR)
```

Comment:
We never clear the backoffDelay, so it might grow indefinitely. Eg. first three attempts fail -> delay is 4s -> one succeeds -> 3 fail and delay is 16s -> etc.

We might want to consider removing "backoffDelay" variable altogether and use something like `delay(DELAY_MILLIS * (max(1, DELAY_FACTOR * retryAttempts)))`. Wdyt?

Reply:
Makes perfect sense. 👍

Reply:
Updated in 83916f7